src/main/java/de/tum/in/www1/artemis/service/AchievementService.java

```diff
@@ -0,0 +1,56 @@
+package de.tum.in.www1.artemis.service;
+
+import java.util.Optional;
+import java.util.Set;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import de.tum.in.www1.artemis.domain.Achievement;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.repository.AchievementRepository;
+import de.tum.in.www1.artemis.repository.UserRepository;
+
+@Service
+public class AchievementService {
+
+    private final AchievementRepository achievementRepository;
+
+    private final UserRepository userRepository;
+
+    public AchievementService(AchievementRepository achievementRepository, UserRepository userRepository) {
+        this.achievementRepository = achievementRepository;
+        this.userRepository = userRepository;
+    }
+
+    public Optional<Achievement> findById(Long achievementId) {
+        return achievementRepository.findById(achievementId);
+    }
+
+    @Transactional
+    public Set<Achievement> findAllForCourse(Long courseId, Long userId) {
```

Comment:
Why did you add here and below `@Transactional`?

Reply:
The users are lazily initialized for achievements, so to be able to get the users within the `hideUsersInAchievements` function, I added that annotation.